### ISSUE TITLE
Shutting down or disabling

### ISSUE THREAD

**Issue by lhunath** [stacksift/SDK#3]
In order to adequately respond to the user's preference, opt-in and opt-out of data collection, it is important that the application is capable of turning off Stackshift as soon as the user revokes their consent for data collection.

There doesn't appear to be a mechanism in place for doing so. Sentry, as an alternative, has the option of passing a `nil` `dsn,` passing `false` to the config's `enabled`, or simply invoking `close()` on the instance. Suggesting the SDK add a method for countering the `start()` method, (eg. `stop()`), as well as a manner for the application to find out whether Stacksift is currently installed and running.

**Comment by mattmassicotte** [comment]
Thanks for making this suggestion. A few others have asked for this as well.

Historically, crash reporters haven't supported a live disable well. It's difficult to restore the global state that crash reporters use. And, because it is global, that restoration process can actually undo other systems' changes. It usually no a problem, but can be a source of very difficult-to-understand bugs.

In fact, I don't think Sentry actually does truly uninstall their crash reporting hooks, but instead just turn them off in some capacity:

https://github.com/getsentry/sentry-cocoa/blob/3a81c428c18790c78258d04f3de07bf3fe758edc/Sources/Sentry/SentryCrashIntegration.m#L148

Regardless, I think supporting this makes sense. It will take quite a bit of work to make the system work correctly in the in-process case. But, we can probably add something fairly simple in Impact that doesn't uninstall, but also won't write out data for crashes.

**Comment by lhunath** [comment]
Very helpful context!

From the perspective of the application, when the concern is rooted in data collection consent, it tends to be sufficient to know that no data is getting persisted or offloaded to remote entities.

I'm not sure if there are different application concerns which might require a full reversal of everything `start` does.  The only thing I can think of is the option of live swapping one crash logging system for another (eg. switching between stacksift and sentry), though I don't know if this is a use case that's as important to support as the data collection one.

**Comment by mattmassicotte** [comment]
I agree. As long as the data doesn't leave the device, it's sufficient. And, now that I've thought about it more, gating the call to `start` should be entirely sufficient. Both in-process and MetricKit reports are never transmitting until the **next** launch.

So, as long as you just check the user preference on launch, and conditionalize the call to `start`, you'll get the exact behavior you want. This is actually essential, even if this API was present. I still think it's a good thing to add, though.
